src/pointerwrappers.jl

```diff
@@ -70,10 +70,12 @@ end
 function Base.setproperty!(pw::PointerWrapper{T}, name::Symbol, v) where T
   i = findfirst(isequal(name), fieldnames(T))
   if isnothing(i)
-    error("type $(string(T)) has no field $name")
+    # For some `struct`s, `fieldnames` gives `data` and not the actual field names, but we can use `Base.setproperty!` for pointers,
+    # see https://github.com/trixi-framework/P4est.jl/issues/72 and https://github.com/trixi-framework/P4est.jl/issues/79
+    Base.setproperty!(pointer(pw), name, v)
+  else
```

Comment:
Iirc, in the other Pr we didn’t have an if else condition but an if with an early return, or am I remembering this incorrectly? Either way you do it, it should imho be consistent between both functions.

Reply:
Done in 1483df31de8a9514610424d2dce3809ae9c53d4a.